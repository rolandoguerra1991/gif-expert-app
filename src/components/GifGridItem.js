import React from 'react'
import PropTypes from 'prop-types';

export default function GifGridItem({ id, title, url }) {
  return (
    <div className='card animate__animated animate__headShake'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};