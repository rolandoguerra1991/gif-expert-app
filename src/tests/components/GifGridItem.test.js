import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
  const image = {
    url: "https://google.com",
    title: "Titulo de la prueba",
    id: "fddsfsdfsdfde3232",
  };
  const wrapper = shallow(<GifGridItem {...image} />);

  test("debe mostrar <GifGridItem /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(image.title);
  });

  test("debe tener una imagen con src y con alt igual a los props", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();

    expect(typeof src).toBe('string');
    expect(typeof alt).toBe('string');
    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });

  test("debe tener la clase animate__headShake", () => {
    const div = wrapper.find("div");
    const { className } = div.props();
    const classes = className.split(' ');
    const exist = classes.includes('animate__headShake');
    expect(exist).toBe(true);
  });
});
