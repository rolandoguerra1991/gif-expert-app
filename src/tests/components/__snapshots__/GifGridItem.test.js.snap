// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__headShake"
>
  <img
    alt="Titulo de la prueba"
    src="https://google.com"
  />
  <p>
    Titulo de la prueba
  </p>
</div>
`;
